src/xhr.ts

```diff
@@ -41,6 +41,51 @@ export class JQueryRequestor extends Requestor {
   }
 }
 
+
+/**
+ * Uses fetch API to make Ajax requests
+ */
+export class FetchRequestor extends Requestor {
+  private rquery = (/\?/);
+
+  xhr<T>(settings: JQueryAjaxSettings): Promise<T> {
+    let url: string = settings.url || '';
+    let requestInit: RequestInit = {};
+    requestInit.method = settings.method;
+    requestInit.mode = 'cors';
+
+    if (settings.data) {
+      if (settings.method && settings.method.toUpperCase() === 'POST') {
+        requestInit.body = <string>settings.data;
+      } else {
+        url += (this.rquery.test(url) ? '&' : '?') + settings.data;
```

Comment:
I am not sure if this is the best way. You are going to have to URL encode this data, otherwise the get request is going to get mangled. We should take a look at what `fetch` provides for `GET` requests, and try and rely on that.

Reply:
You're right, I should have encoded the keys and values with [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent). 
As far as I know, `fetch` doesn't provide anything in particular for sending data in `GET` requests. According to the [spec](https://fetch.spec.whatwg.org/#fetch-api), the standard way of managing request params looks as follows:

````
var url = new URL("https://geo.example.org/api"),
    params = {lat:35.696233, long:139.570431}
Object.keys(params).forEach(key => url.searchParams.append(key, params[key]))
fetch(url).then(/* … */)
````
I thought about using `URLSearchParams` at first but I didn't go this way [because of the browser support](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams#Browser_compatibility). I've changed it anyway because looking at the code, I don't find any point where a `Requestor` is being used for sending data via `GET`. I wonder if we should even allow to send data this way taking into account that:

- The `openid` spec says that the [token request](https://openid.net/specs/openid-connect-core-1_0.html#TokenRequest) must send data by `POST`. 

- The [token revokation](https://tools.ietf.org/html/rfc7009#section-2) request must be a `POST` request. 

- The [discovery request](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest) is a `GET` request that doesn't send any data.

Reply:
You are right. We probably don't need to support `GET`. 

Your PR looks good.  Do you want to rebase your changes with mine, and resend them to me ? Also could you please sign the OpenID CLA?

Reply:
Thanks @tikurahul. I've signed the OpenID CLA and rebased your changes.